### PR TITLE
[JENKINS-69110] Plugin manager "select all" checkbox doesn't change state on click

### PR DESCRIPTION
**Pull request by janfaracik**
See [JENKINS-69110](https://issues.jenkins.io/browse/JENKINS-69110).

Not too sure how I'd have missed this before merge - sorry about that! Should all be working now.

### Proposed changelog entries

* Change state of plugin manager "select all" checkbox when it is clicked.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@NotMyFault @jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6911"></a>

### CODE REVIEW

**Comment by NotMyFault** [comment]
Thanks for addressing, but it looks like that mitigates the error only, without restoring the chevron shown on https://user-images.githubusercontent.com/43062514/177877649-53f09952-0736-4e9c-a089-2eb98e194281.mov

**Comment by janfaracik** [comment]
> Thanks for addressing, but it looks like that mitigates the error only, without restoring the chevron shown on https://user-images.githubusercontent.com/43062514/177877649-53f09952-0736-4e9c-a089-2eb98e194281.mov, no?

Dang GitHub videos never load properly for me. I'm hoping you mean the chevron with the options in? E.g. All, Compatible, None? The dropdown will only show if you have incompatible plugins in the list.

File: war/src/main/js/components/row-selection-controller/index.js
Comment (timja) [inline war/src/main/js/components/row-selection-controller/index.js:49]: Likely needs a null check I don’t think this works with htmlunit

Comment (janfaracik) [reply war/src/main/js/components/row-selection-controller/index.js:49]: Should be okay as we're already using optional chaining in the file 🤞

**Comment by NotMyFault** [comment]
> Dang GitHub videos never load properly for me. I'm hoping you mean the chevron with the options in? E.g. All, Compatible, None?  [#6806 (comment)](https://github.com/jenkinsci/jenkins/pull/6806#issuecomment-1180632781).

Yeah

> The dropdown will only show if you have incompatible plugins in the list

Oic, makes sense.

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Comment by daniel-beck** [comment]
When the page loads without available updates, I get this in the console:

```
Uncaught TypeError: moreOptionsButton is null
    <anonymous> index.js:13
    <anonymous> index.js:3
    <anonymous> row-selection-controller.js:71
```

Probably best to clean this up as well?

File: war/src/main/js/components/row-selection-controller/index.js
Comment (NotMyFault) [inline war/src/main/js/components/row-selection-controller/index.js:13]: Optional chaining with assignments would be a dream 😢

Comment (timja) [reply war/src/main/js/components/row-selection-controller/index.js:13]:
```
 Invalid left-hand side in assignment expression. (13:4)
```

Comment (janfaracik) [reply war/src/main/js/components/row-selection-controller/index.js]: Dang, didn't realise you couldn't do that. I've changed it - cheers :)

**Pull request merged**
